controllers/humiocluster_controller.go

```diff
@@ -2349,3 +2363,80 @@ func getHumioNodePoolManagers(hc *humiov1alpha1.HumioCluster) HumioNodePoolList
 	}
 	return humioNodePools
 }
+
+// podLabelsForHumio returns the labels for selecting the resources
+// belonging to the given humioCluster CR name.
+func (r *HumioClusterReconciler) podLabelsForHumio(name string) map[string]string {
+	return map[string]string{"app": "humio", "humio_cr": name}
+}
+
+func (r *HumioClusterReconciler) reconcilePodDisruptionBudget(ctx context.Context, humioCluster *humiov1alpha1.HumioCluster) error {
+	// Define the desired PodDisruptionBudget object
+	pdb := &policyv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      humioCluster.Name + "-pdb", // Or a more suitable name
+			Namespace: humioCluster.Namespace,
+			Labels:    r.podLabelsForHumio(humioCluster.Name), // Make sure labels are correct
+		},
+		Spec: policyv1.PodDisruptionBudgetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: r.podLabelsForHumio(humioCluster.Name),
+			},
+		},
+	}
+
+	// Set the MinAvailable or MaxUnavailable value
+	if humioCluster.Spec.PodDisruptionBudget != nil {
+		if humioCluster.Spec.PodDisruptionBudget.MinAvailable != nil {
+			pdb.Spec.MinAvailable = humioCluster.Spec.PodDisruptionBudget.MinAvailable
+		} else if humioCluster.Spec.PodDisruptionBudget.MaxUnavailable != nil {
+			pdb.Spec.MaxUnavailable = humioCluster.Spec.PodDisruptionBudget.MaxUnavailable
+		}
+	} else {
+		// Set default values if not specified in the CR
+		defaultMinAvailable := intstr.FromInt(2) // Example default: at least 2 pods available
+		pdb.Spec.MinAvailable = &defaultMinAvailable
+	}
+
+	// Check if the PodDisruptionBudget already exists
+	foundPdb := &policyv1.PodDisruptionBudget{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: pdb.Name, Namespace: pdb.Namespace}, foundPdb)
+	if err != nil && k8serrors.IsNotFound(err) {
```

Comment:
I would streamline the error check like this:
```
	if err != nil {
		return err
	}
	if k8serrors.IsNotFound(err) {
		// Create the PodDisruptionBudget
		r.Log.Info("Creating a new PodDisruptionBudget", "PDB.Namespace", pdb.Namespace, "PDB.Name", pdb.Name)
		err = r.Client.Create(ctx, pdb)
		if err != nil {
			return err
		}
		return nil
	} 
```

Reply:
@bsocaciu  I  feel that your suggestion has a flaw which the first of statement would return all errors before reaching the second if statement. The second if would never be reached because err is already handled.

Reply:
Oh, yeah. I saw now you're actually handling it there. My bad 😁